tasks/components/modals/ConfirmDialog.tsx

```diff
@@ -0,0 +1,74 @@
+import { tw } from '@helpwave/common/twind/index'
+import type { PropsWithChildren } from 'react'
+import { Modal } from './Modal'
+import { Button } from '../Button'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+
+type ConfirmDialogTranslation = {
+  confirm: string,
+  cancel: string,
+  decline: string
+}
+
+const defaultConfirmDialogTranslation = {
+  en: {
+    confirm: 'Confirm',
+    cancel: 'Cancel',
+    decline: 'Decline'
+  },
+  de: {
+    confirm: 'Bestätigen',
+    cancel: 'Abbrechen',
+    decline: 'Ablehnen'
+  }
+}
+
+type ConfirmDialogProps = {
+  isOpen: boolean,
+  title?: string,
+  description?: string,
+  isShowingDecline?: string,
+  onClose: (operation: 'confirm' | 'decline' | 'closed') => void
+}
+export const ConfirmDialog = ({
+  language,
+  children,
+  isOpen,
+  title,
+  description,
+  isShowingDecline,
+  onClose
+}: PropsWithLanguage<ConfirmDialogTranslation, PropsWithChildren<ConfirmDialogProps>>) => {
+  const translation = useTranslation(language, defaultConfirmDialogTranslation)
+  return (
+    <Modal isOpen={isOpen} title={title} description={description} onClose={event => {
+      event.stopPropagation()
+      onClose('closed')
+    }}>
+      {children}
+      <div className={tw('flex flex-row mt-3 gap-x-4 justify-end')}>
+        <Button color="accent" onClick={event => {
+          event.stopPropagation()
+          onClose('closed')
+        }}>
+          {translation.cancel}
+        </Button>
+        {isShowingDecline && (
+          <Button color="negative" onClick={event => {
+            event.stopPropagation()
+            onClose('decline')
```

Comment:
I'd vouch for having individual callbacks for each action (`'onDecline'`, `onAccept` or `onConfirm`, `onClose`)

But I'm open to discussing this; if there is something specific this allows I'm all for it